include/exadg/aero_acoustic/volume_coupling.h

```diff
@@ -0,0 +1,123 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2023 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef INCLUDE_EXADG_AERO_ACOUSTIC_VOLUME_COUPLING_H_
+#define INCLUDE_EXADG_AERO_ACOUSTIC_VOLUME_COUPLING_H_
+
+#include <exadg/aero_acoustic/calculators/source_term_calculator.h>
+#include <exadg/aero_acoustic/single_field_solvers/acoustics.h>
+#include <exadg/aero_acoustic/single_field_solvers/fluid.h>
+#include <exadg/aero_acoustic/user_interface/parameters.h>
+
+namespace ExaDG
+{
+namespace AeroAcoustic
+{
+/**
+ * A class that handles the volume coupling between fluid and acoustic.
+ */
+template<int dim, typename Number>
+class VolumeCoupling
```

Comment:
Is this class intended to handle both directions? Will you add another function `acoustic_to_fluid()` later? (I somewhat deduce this from the naming `fluid_to_acoustic`.)

If this is the case, maybe this is something to discuss in more detail, so that I better understand the motivation behind. From a software design perspective, I would currently argue to aim at a design as modular as possible, which could lead to a solution where the different coupling directions are realized independently.

Reply:
Yes, the class is intended to be used for both directions. The advantage to keep both directions in the same class is that, e.g. the setup of the classes from `deal.II` (such as the non-nested multigrid class or others like a method where the meshes are intersected) are quite expensive and can be used to transfer vectors in both directions. In the present example, one could use the `prolongate_and_add()` method for acoustic->fluid coupling. 

Besides that, I think it is natural to handle the transfer of solutions in both directions in one class since it is similar to the multigrid classes in `deal.II`.